semantic-python/src/Language/Python/Core.hs

```diff
@@ -215,33 +215,34 @@ instance Compile Py.ExecStatement
 deriving instance Compile Py.Expression
 
 instance Compile Py.ExpressionStatement where
-  compileCC it@Py.ExpressionStatement
+  compile it@Py.ExpressionStatement
     { Py.extraChildren = children
     } cc = do
-    foldr compileCC cc children >>= locate it
+    foldr compile cc children >>= locate it
 
 instance Compile Py.ExpressionList where
-  compileCC it@Py.ExpressionList { Py.extraChildren = [child] } cc
-    = compileCC child cc >>= locate it
-  compileCC Py.ExpressionList { Py.extraChildren = items } _
+  compile it@Py.ExpressionList { Py.extraChildren = [child] } cc
+    = compile child cc >>= locate it
+  compile Py.ExpressionList { Py.extraChildren = items } _
     = fail ("unimplemented: ExpressionList of length " <> show items)
 
 
 instance Compile Py.False where
-  compileCC it _ = locate it $ bool False
+  compile it _ = locate it $ bool False
 
 instance Compile Py.Float
 instance Compile Py.ForStatement
 
 instance Compile Py.FunctionDefinition where
-  compileCC it@Py.FunctionDefinition
+  compile it@Py.FunctionDefinition
     { name       = Py.Identifier _ann1 name
     , parameters = Py.Parameters _ann2 parameters
     , body
     } cc = do
       -- Compile each of the parameters, then the body.
       parameters' <- traverse param parameters
-      body' <- compile body
+      -- BUG: ignoring the continuation here
+      body' <- compile body (pure none)
```

Comment:
This _might_ be correct. The continuation of a function definition should not be passed the compilation of the body, but rather the compilation of the entire function. (I think?)